### ISSUE TITLE
Array Shuffle() in C# HTML5 fails with no evident error. 

### ISSUE THREAD

**Issue by grazianobolla** [godotengine/godot#60840]
### Godot version

3.4.4.stable.mono.official

### System information

Ubuntu 22.04 GTX1060 I58400 16GB-RAM GLES3

### Issue description

The Shuffle() function used to randomize an Array object doesent work with c# on HTML5

### Steps to reproduce

Create a new project, add a script to any node, and test this:

```csharp
public override void _Ready()
{
    Array<int> arr = new Array<int>();
    arr.Add(3);
    arr.Add(1);
    arr.Add(4);
    arr.Shuffle();
    GD.Print(arr);
}
```

Try running it on the browser via export or by clicking "Run in Browser" in Godot.

### Minimal reproduction project

[testing-sandbox.zip](https://github.com/godotengine/godot/files/8644389/testing-sandbox.zip)

**Comment by grazianobolla** [comment]
I forgot to mention that the message error that appears says:

`null function or function signature mismatch`

But there is not any relevant information to track, I spent 1 hour until I found what was causing the problem.

**Comment by MunsterPlop** [comment]
I encountered this bug while doing Ludum Dare and I can confirm this is still relevant as of 3.5.1.

I don't know if it could be of any help but I'm seeing this in the browser inspector:

`
exception thrown: RuntimeError: indirect call signature mismatch,do_icall@:wasm-function[37148]:0x86cee9
do_icall_wrapper@:wasm-function[10678]:0x201bde
interp_exec_method_full@:wasm-function[4640]:0xe2b16
interp_runtime_invoke@:wasm-function[33852]:0x7b81f3
mono_jit_runtime_invoke@:wasm-function[32889]:0x77a30c
do_runtime_invoke@:wasm-function[6708]:0x15598e
mono_runtime_try_invoke@:wasm-function[2678]:0x87d12
mono_runtime_invoke@:wasm-function[5691]:0x128b87
GDMonoUtils::runtime_invoke(_MonoMethod*, void*, void**, _MonoException**)@:wasm-function[7461]:0x173945
GDMonoMethod::invoke(_MonoObject*, Variant const**, _MonoException**) const@:wasm-function[6330]:0x14577f
CSharpInstance::_call_multilevel(_MonoObject*, StringName const&, Variant const**, int)@:wasm-function[74752]:0xf6c363
CSharpInstance::call_multilevel(StringName const&, Variant const**, int)@:wasm-function[74749]:0xf6c026
CSharpInstance::call_multilevel_reversed(StringName const&, Variant const**, int)@:wasm-function[74748]:0xf6bfd3
Node::_notification(int)@:wasm-function[20742]:0x3f9226
Node::_notificationv(int, bool)@:wasm-function[1668]:0x6746a
Object::notification(int, bool)@:wasm-function[1174]:0x4d625
Node::_propagate_ready()@:wasm-function[20735]:0x3f8560
Node::_set_tree(SceneTree*)@:wasm-function[7147]:0x16a974
Node::_add_child_nocheck(Node*, StringName const&)@:wasm-function[20745]:0x3f9760
Node::add_child(Node*, bool)@:wasm-function[1016]:0x41a99
SceneTree::_change_scene(Node*)@:wasm-function[46493]:0xad03bb
MethodBind1<Node*>::call(Object*, Variant const**, int, Variant::CallError&)@:wasm-function[58513]:0xd2884c
Object::call(StringName const&, Variant const**, int, Variant::CallError&)@:wasm-function[8932]:0x1bb880
MessageQueue::_call_function(Object*, StringName const&, Variant const*, int, bool)@:wasm-function[68830]:0xded904
MessageQueue::flush()@:wasm-function[4864]:0xf8702
SceneTree::idle(float)@:wasm-function[46456]:0xace647
Main::iteration()@:wasm-function[69779]:0xe2cdd5
OS_JavaScript::main_loop_iterate()@:wasm-function[51449]:0xbf8033
main_loop_callback()@:wasm-function[16175]:0x2fa49b
wasm_invoke_v@:wasm-function[27300]:0x4d5ddc
browserIterationFunc@http://localhost:8060/tmp_js_export.js:9:117405
runIter@http://localhost:8060/tmp_js_export.js:9:120137
Browser_mainLoop_runner@http://localhost:8060/tmp_js_export.js:9:118620
[tmp_js_export.js:369:18](http://localhost:8060/tmp_js_export.js)
Uncaught RuntimeError: indirect call signature mismatch
    browserIterationFunc http://localhost:8060/tmp_js_export.js:9
    runIter http://localhost:8060/tmp_js_export.js:9
    Browser_mainLoop_runner http://localhost:8060/tmp_js_export.js:9
`

**Comment by akien-mga** [comment]
Fixed by #68252.
